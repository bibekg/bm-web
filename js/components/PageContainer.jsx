// @flow

import styled from 'styled-components'
import { breakpoints, colors } from 'styles'
import dotty from 'dotty'

type ContainerOptionsType = {
  maxWidth?: { small?: string, large?: string },
  noBackground: ?boolean
}

const DEFAULT_MAX_WIDTH_LARGE = `${breakpoints.medium}px`
const DEFAULT_MAX_WIDTH_SMALL = '90%'

export default (userOptions?: ContainerOptionsType): * => {
  const options = {
    maxWidth: {
      small: dotty.get(userOptions, 'maxWidth.small') || DEFAULT_MAX_WIDTH_SMALL,
      large: dotty.get(userOptions, 'maxWidth.large') || DEFAULT_MAX_WIDTH_LARGE
    },
    noBackground: dotty.exists(userOptions, 'noBackground') ? dotty.get(userOptions, 'noBackground') : false
  }

  return styled.div`
    width: 100%;
    min-height: 100vh;
    /* Default to pale blue background for pages */
    ${options.noBackground ? '' : `background-color: ${colors.paleBlue}`};

    /* Large screens */
    @media (min-width: ${breakpoints.medium}px) {
      padding: 50px 20px;
      & > * {
        max-width: ${options.maxWidth.large};
      }
    }

    /* Small screens */
    @media (max-width: ${breakpoints.medium - 1}px) {
      padding: 25px 10px;
      & > * {
        max-width: ${options.maxWidth.small};
      }
    }
  `
}
