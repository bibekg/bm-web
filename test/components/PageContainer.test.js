import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import PageContainer from 'components/PageContainer'

test('PageContainer optionless render', () => {
  const PageContainerElement = PageContainer()
  const tree = renderer.create(<PageContainerElement />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('PageContainer with options', () => {
  const PageContainerElement = PageContainer({
    noBackground: true,
    maxWidth: {
      large: '500px',
      small: '20%'
    }
  })
  const tree = renderer.create(<PageContainerElement />).toJSON()
  expect(tree).toMatchSnapshot()
})
