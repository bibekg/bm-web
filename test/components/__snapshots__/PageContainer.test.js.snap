// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer optionless render 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
  background-color: #dbedff;
}

@media (min-width: 768px) {
  .c0 {
    padding: 50px 20px;
  }

  .c0 > * {
    max-width: 768px;
  }
}

@media (max-width: 767px) {
  .c0 {
    padding: 25px 10px;
  }

  .c0 > * {
    max-width: 90%;
  }
}

<div
  className="c0"
/>
`;

exports[`PageContainer with options 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
}

@media (min-width: 768px) {
  .c0 {
    padding: 50px 20px;
  }

  .c0 > * {
    max-width: 500px;
  }
}

@media (max-width: 767px) {
  .c0 {
    padding: 25px 10px;
  }

  .c0 > * {
    max-width: 20%;
  }
}

<div
  className="c0"
/>
`;
